internal/configsource/fileconfigsource/README.md

```diff
@@ -0,0 +1,42 @@
+# File Config Source (Alpha)
+
+Use the file config source to inject YAML fragments or scalars into the
+configuration.
+
+## Configuration
+
+Under the `config_sources:` use `file:` or `file/<name>:` to create a
+file config source. The following parameters are available to
+customize file config sources:
+
+```yaml
+config_sources:
+  file:
+```
+
+By default, the config source will monitor for updates on the used files
+and will trigger a configuration reload when they are updated. Optionally,
+the config source can be configured to delete the injected file (typically
```

Comment:
Mind specifying when the file is deleted?  Appears to be after reading.

Reply:
Added a blurb for that.